### ISSUE TITLE
Expired https certificate

### ISSUE THREAD

**Issue by tjayrush** [Great-Hill-Corporation/trueblocks-manage#90]
This email chain describes a problem we've each year when the certificate expires. It would be helpful if we could have instructions to fix it in this issue:

```
On Jan 18, 2019, at 5:07 PM, Edward Mazurek <changes@gmail.com> wrote:

No, this is due to an SSL challenge method being retired. We should be good now, with the challenge method and auto renew as well.

Ed

On Fri, Jan 18, 2019, 4:03 PM Thomas Jay Rush <jrush@quickblocks.io wrote:
Is this the same thing that happened before?

-------------------------------
Thomas Jay Rush
QuickBlocks.io
jrush@quickblocks.io
(610) 519-9413

> On Jan 18, 2019, at 3:47 PM, Edward Mazurek <changes@gmail.com> wrote:
> 
> Took care of this. All set.
> 
> Ed
> 
> On Fri, Jan 18, 2019 at 12:00 PM Edward Mazurek <changes@gmail.com> wrote:
> Thanks Vidal. I help Jay with some of his websites. Thanks for the recommended action. I'll take the actions you prescribed.
> 
> Ed
> 
> On Fri, Jan 18, 2019, 10:52 AM Vidal Herrero <vidal.herrero@gmail.com wrote:
> Hello
> 
> I remember I configured certificate to be autorenewed, bnut it seems to be a problem.
> 
> Fortunately I still have the acces to your server 
> 
> I hace executed:
> 
> -----------------------------------
> 
> certbot renew --dry-run
> 
> --------------------------------------
> 
> and it displays a problem:
> 
> -------------------------------
> 
> "Attempting to renew cert (quickblocks.io) from /etc/letsencrypt/renewal/quickblocks.io.conf produced an unexpected error: Client with the currently selected authenticator does not support any combination of challenges that will satisfy the CA.. Skipping."
> ---------------------------
> 
> I have  executed:
> 
> root@wordpress-1gb-nyc3-01:/etc/apache2/sites-available# certbot --version
> certbot 0.19.0
> 
> 
> 
> Last certbot Version is 0.29 I guess
> 
> ---
> 
> root@wordpress-1gb-nyc3-01:/etc/apache2/sites-available# certbot certificates
> 
> -------------------------------------------------------------------------------
> Found the following certs:
>   Certificate Name: quickblocks.io
>     Domains: www.quickblocks.io quickblocks.io
>     Expiry Date: 2018-12-12 20:40:22+00:00 (INVALID: EXPIRED)
>     Certificate Path: /etc/letsencrypt/live/quickblocks.io/fullchain.pem
>     Private Key Path: /etc/letsencrypt/live/quickblocks.io/privkey.pem
>   Certificate Name: quickblocks.io-0001
>     Domains: quickblocks.io
>     Expiry Date: 2019-04-18 14:38:25+00:00 (VALID: 89 days)
>     Certificate Path: /etc/letsencrypt/live/quickblocks.io-0001/fullchain.pem
>     Private Key Path: /etc/letsencrypt/live/quickblocks.io-0001/privkey.pem
> -------------------------------------------------------------------------------
> 
> 
> Yyou have 2 certificates to the same domain.
> 
> 
> 
> __________________
> 
> I recommend :
> 
> 1.- update certbot to the last version 
> 
> 2.- unistall both certificates, 
> 
> 3.- Install correctly a third certificate for quickblocks.io  and www.quicblocks.io
> 
> 4.- configure automatically renew 
> 
> 
> 
> let me know waht you think
> 
> 
> 
> 
> Best Regards
> 
> Vidal
> 
> 
> 
> 
> 
> 
> 
> El 18/01/2019 a las 15:06, Thomas Jay Rush escribió:
>> Hi Vidal. Hi Ed.
>> 
>> Vidal,
>> 
>> I think this is the free certificate you setup when you built my website. Is there any way to set this up permanently so I don’t get these notices every three months?
>> 
>> Ed,
>> 
>> I copied you in, so you can see what Vidal says to help here.
>> 
>> 
>> -------------------------------
>> Thomas Jay Rush
>> QuickBlocks.io
>> 
>> jrush@quickblocks.io
>> 
>> (610) 519-9413
>> 
>> 
>>> On Jan 18, 2019, at 5:33 AM, <noreply@letsencrypt.org> <noreply@letsencrypt.org>
>>>  wrote:
>>> 
>>> Hello,
>>> 
>>> **Action is required to prevent your Let's Encrypt certificate renewals from breaking.**
>>> 
>>> Your Let’s Encrypt client used ACME TLS-SNI-01 domain validation to issue a certificate in the past 60 days.
>>> 
>>> TLS-SNI-01 validation is reaching end-of-life and will stop working on **February 13th, 2019.**
>>> 
>>> You need to update your ACME client to use an alternative validation method (HTTP-01, DNS-01 or TLS-ALPN-01) before this date or your certificate renewals will break and existing certificates will start to expire.
>>> 
>>> If you need help updating your ACME client, please open a new topic in the Help category of the Let's Encrypt community forum:
>>> 
>>>  
>>> https://community.letsencrypt.org/c/help
>>> 
>>> 
>>> Please answer all of the questions in the topic template so we can help you.
>>> 
>>> For more information about the TLS-SNI-01 end-of-life please see our API announcement:
>>> 
>>> 
>>> https://community.letsencrypt.org/t/february-13-2019-end-of-life-for-all-tls-sni-01-validation-support/74209
>>> 
>>> 
>>> Thank you,
>>>  Let's Encrypt Staff
>>> 
>>> 


```

**Comment by wildmolasses** [comment]
Hi Jay. You shouldn't have this problem anymore. The certificates should be set to auto-renew. The last time we had this problem, it was due to an unrelated problem. The problem was that the SSL challenge method we relied on to renew the SSL certificates was being retired. So we had to switch to a new SSL challenge method. We did that and now the certificates should auto-renew.

But... in the event that a certificate doesn't renew, then the command would be `sudo certbot renew`.

**Comment by wildmolasses** [comment]
"is it configured to auto-renew?" yes, you can see by typing `systemctl list-timers`. The following should appear.

![Screen Shot 2020-02-26 at 12 13 47 AM](https://user-images.githubusercontent.com/5423313/75317195-f7f42780-582c-11ea-84dc-f59d269e6208.png)

**Comment by tjayrush** [comment]
I ran `sudo certbot renew` and got this result:

```
Saving debug log to /var/log/letsencrypt/letsencrypt.log

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Processing /etc/letsencrypt/renewal/quickblocks.io.conf
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Cert is due for renewal, auto-renewing...
Plugins selected: Authenticator apache, Installer apache
Renewing an existing certificate
Performing the following challenges:
http-01 challenge for quickblocks.io
http-01 challenge for www.quickblocks.io
Waiting for verification...
Cleaning up challenges
Attempting to renew cert (quickblocks.io) from /etc/letsencrypt/renewal/quickblocks.io.conf produced an unexpected error: Failed authorization procedure. quickblocks.io (http-01): urn:ietf:params:acme:error:unauthorized :: The client lacks sufficient authorization :: Invalid response from https://quickblocks.io/.well-known/acme-challenge/GL2M4A-ieZfFn3OES7uJ-4wMyx1Mdxy07ROHp0sez7A [104.236.95.164]: "<!DOCTYPE html>\n<!--[if IE 6]>\n<html id=\"ie6\" lang=\"en-US\" prefix=\"og: http://ogp.me/ns#\">\n<![endif]-->\n<!--[if IE 7]>\n<html id=", www.quickblocks.io (http-01): urn:ietf:params:acme:error:unauthorized :: The client lacks sufficient authorization :: Invalid response from https://quickblocks.io/.well-known/acme-challenge/_p_ckzc9A_jLvVNQ4x237JDcDR5lXkTz6PuLvor35-8 [104.236.95.164]: "<!DOCTYPE html>\n<!--[if IE 6]>\n<html id=\"ie6\" lang=\"en-US\" prefix=\"og: http://ogp.me/ns#\">\n<![endif]-->\n<!--[if IE 7]>\n<html id=". Skipping.

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Processing /etc/letsencrypt/renewal/trueblocks.io.conf
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Attempting to parse the version 1.1.0 renewal configuration file found at /etc/letsencrypt/renewal/trueblocks.io.conf with version 0.31.0 of Certbot. This might not work.
Cert not yet due for renewal

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Processing /etc/letsencrypt/renewal/www.quickblocks.io.conf
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Attempting to parse the version 1.1.0 renewal configuration file found at /etc/letsencrypt/renewal/www.quickblocks.io.conf with version 0.31.0 of Certbot. This might not work.
Cert not yet due for renewal

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Processing /etc/letsencrypt/renewal/www.trueblocks.io.conf
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Attempting to parse the version 1.1.0 renewal configuration file found at /etc/letsencrypt/renewal/www.trueblocks.io.conf with version 0.31.0 of Certbot. This might not work.
Cert not yet due for renewal
All renewal attempts failed. The following certs could not be renewed:
 /etc/letsencrypt/live/quickblocks.io/fullchain.pem (failure)

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

The following certs are not due for renewal yet:
 /etc/letsencrypt/live/trueblocks.io/fullchain.pem expires on 2020-04-21 (skipped)
 /etc/letsencrypt/live/www.quickblocks.io/fullchain.pem expires on 2020-04-21 (skipped)
 /etc/letsencrypt/live/www.trueblocks.io/fullchain.pem expires on 2020-04-21 (skipped)
All renewal attempts failed. The following certs could not be renewed:
 /etc/letsencrypt/live/quickblocks.io/fullchain.pem (failure)
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
1 renew failure(s), 0 parse failure(s)
IMPORTANT NOTES:
- The following errors were reported by the server:

  Domain: quickblocks.io
  Type:   unauthorized
  Detail: Invalid response from
  https://quickblocks.io/.well-known/acme-challenge/GL2M4A-ieZfFn3OES7uJ-4wMyx1Mdxy07ROHp0sez7A
  [104.236.95.164]: "<!DOCTYPE html>\n<!--[if IE 6]>\n<html
  id=\"ie6\" lang=\"en-US\" prefix=\"og:
  http://ogp.me/ns#\">\n<![endif]-->\n<!--[if IE 7]>\n<html id="

  Domain: www.quickblocks.io
  Type:   unauthorized
  Detail: Invalid response from
  https://quickblocks.io/.well-known/acme-challenge/_p_ckzc9A_jLvVNQ4x237JDcDR5lXkTz6PuLvor35-8
  [104.236.95.164]: "<!DOCTYPE html>\n<!--[if IE 6]>\n<html
  id=\"ie6\" lang=\"en-US\" prefix=\"og:
  http://ogp.me/ns#\">\n<![endif]-->\n<!--[if IE 7]>\n<html id="

  To fix these errors, please make sure that your domain name was
  entered correctly and the DNS A/AAAA record(s) for that domain
  contain(s) the right IP address.
```

**Comment by tjayrush** [comment]
For `systemctl list-timers`, I get:

```
NEXT                         LEFT          LAST                         PASSED       UNIT                         ACTIVATES
Wed 2020-02-26 11:10:51 EST  1h 31min left Wed 2020-02-26 05:05:37 EST  4h 34min ago apt-daily.timer              apt-daily.service
Wed 2020-02-26 15:21:24 EST  5h 41min left Tue 2020-02-25 15:21:24 EST  18h ago      systemd-tmpfiles-clean.timer systemd-tmpfiles-clean.service
Wed 2020-02-26 18:19:05 EST  8h left       Wed 2020-02-26 08:15:14 EST  1h 24min ago certbot.timer                certbot.service
Thu 2020-02-27 06:44:57 EST  21h left      Wed 2020-02-26 06:58:16 EST  2h 41min ago apt-daily-upgrade.timer      apt-daily-upgrade.service
```

**Comment by wildmolasses** [comment]
I believe I debugged the issue. Give this a quick read and see if it makes sense. All was well with the certificates. They were generated with a >1.0.0 version of certbot (formerly at `~/certbot-auto`). Then, at some point, an old version of certbot was ran. This version was at `/usr/bin/certbot`. I *believe* that running this invalidated the previous valid certificates. So anyway. I deleted the old version of certbot and replaced it with the new one (I moved the new version from `~/certbot-auto` to`/usr/bin/certbot`). Now there is no trouble generating the certificates, and there are no errors.
